src/lava/lib/optimization/solvers/generic/builder.py

```diff
@@ -0,0 +1,365 @@
+    MacroStateReader, VariablesImplementation
+from lava.lib.optimization.solvers.generic.hierarchical_processes import \
+    ContinuousVariablesProcess, CostConvergenceChecker, \
+    DiscreteVariablesProcess, SatConvergenceChecker
+from lava.lib.optimization.solvers.generic.monitoring_processes \
+    .solution_readout.process import SolutionReadout
+from lava.magma.core.model.model import AbstractProcessModel
+from lava.magma.core.model.sub.model import AbstractSubProcessModel
+from lava.magma.core.process.interfaces import AbstractProcessMember
+from lava.magma.core.process.ports.ports import InPort
+from lava.magma.core.process.process import AbstractProcess, LogConfig
+from lava.magma.core.process.variable import Var
+from lava.magma.core.resources import AbstractComputeResource
+from lava.magma.core.sync.protocol import AbstractSyncProtocol
+from lava.proc.dense.process import Dense
+from lava.proc.read_gate.process import ReadGate
+from numpy import typing as npt
+
+
+class SolverProcessBuilder:
+    """Builder to dynamically create the process and model for the solver of an
+    optimization problem."""
+
+    def __init__(self):
+        self._process_constructor = None
+        self._model_constructor = None
+        self._process = None
+        self._model = None
+
+    @property
+    def solver_process(self) -> AbstractProcess:
+        """Returns the solver process if already created."""
+        self.verify_process_exists()
+        return self._process
+
+    @property
+    def solver_model(self) -> AbstractProcessModel:
+        """Returns the solver process model if already created."""
+        self.verify_model_exists()
+        return self._model
+
+    def verify_process_exists(self):
+        """Assert the solver process has already been created."""
+        msg = """Process has not been created yet. Make sure the
+        create_solver_process method was already called."""
+        assert self._process is not None, msg
+
+    def verify_model_exists(self):
+        """Assert the solver process model has already been created."""
+        msg = """Process model has not been created yet. Make sure the
+        create_solver_model method was already called."""
+        assert self._model is not None, msg
+
+    def create_solver_process(self,
+                              problem: OptimizationProblem,
+                              hyperparameters: ty.Dict[str, ty.Union[int,
+                                                                     npt.ArrayLike]]):
+        """Create and set a solver process for the specified optimization
+        problem.
+
+        Parameters
+        ----------
+        problem: Optimization problem defined by cost and constraints which
+        will be used to ensemble the necessary variables and ports with their
+        shape and initial values deriving from the problem specification.
+        hyperparameters: A dictionary specifying values for steps_to_fire,

```

Comment:
Users will need a bit more information about the hyperparameters.

Reply:
At least you should point to the tutorial, where the hyperparameters are defined in a table